homeassistant/components/sma/sensor.py

```diff
@@ -84,7 +84,11 @@ def __init__(
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return self._sensor.name
+        name_prefix = "SMA"
+        if self._attr_device_info is not None:
+            name_prefix = str(self._attr_device_info.get("name", "SMA"))
```

Comment:
casting to `str` to prevent 
`homeassistant/components/sma/sensor.py:89: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")  [assignment]`

Is there a better way to do this?

Reply:
Yes,  I'll do a suggestion (with spaces this time 😬 )